core/domain/email_services.py

```diff
@@ -121,9 +123,10 @@ def _is_sender_email_valid(sender_email):
         bool. Whether the sender_email is valid.
     """
     # Regex for 'SENDER_NAME <SENDER_EMAIL_ADDRESS>' or 'email_address'.
+    # Matches a name and then an email in the angle brackets.
+    # Email regex explained in _is_email_valid().
     sender_email_regex = (
-        r'^[a-zA-Z._][a-zA-Z._ ]* <.+@(\[?)[a-zA-Z0-9-.]+.([a-zA-Z]{2,3}|' +
-        r'[0-9]{1,3})(]?)>$')
+        r'^[a-zA-Z._][a-zA-Z._ ]* <.+@[a-zA-Z0-9-.]+\.([a-zA-Z]+|[0-9]+)>$')
```

Comment:
Try to remove the duplication between this regex and the above one -- some parts are reused.